sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -206,7 +206,11 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
    * for visualization.
    */
   protected final def executeQuery[T](query: => T): T = {
-    RDDOperationScope.withScope(sparkContext, nodeName, false, true) {
+    val nodeNameScope = this match {
```

Comment:
How about overriding the `nodeName` of `WholeStageCodegenExec`.
I think it is hacky to make special handling here.

Reply:
ah good idea! we can also remove the hack in SPARK-29894

Reply:
+1